scipy/stats/tests/test_morestats.py

```diff
@@ -686,6 +687,85 @@ def binom_test_func(x, n=None, p=0.5, alternative='two-sided'):
 class TestBinomTest:
     """Tests for stats.binomtest."""
 
+    # Expected results here are from old binom_test function.
+    # The alternative will be left unspecified which defaults
+    # it to two-sided.
+    @pytest.mark.xfail_on_32bit("The large inputs make these tests "
+                                "sensitive to machine epsilon level")
+    def test_two_sided_pvalues1(self):
+        # These tests work on all OS's but fail on
+        # Linux_Python_37_32bit_full due to numerical issues caused
+        # by large inputs.
+        res = stats.binomtest(10079999, 21000000, 0.48)
+        assert_allclose(res.pvalue, 0.979042561004596, rtol=1e-15)
+        res = stats.binomtest(10079990, 21000000, 0.48)
+        assert_allclose(res.pvalue, 0.9785298857599378, rtol=1e-15)
+        res = stats.binomtest(10080009, 21000000, 0.48)
+        assert_allclose(res.pvalue, 0.9786038762958954, rtol=1e-15)
+        res = stats.binomtest(10080017, 21000000, 0.48)
+        assert_allclose(res.pvalue, 0.9778567637538729, rtol=1e-15)
+
+    def test_two_sided_pvalues2(self):
+        res = stats.binomtest(9, n=21, p=0.48)
+        assert_allclose(res.pvalue, 0.6689672431938848, rtol=1e-15)
```

Comment:
this one is failing too (similar to other failure in edge cases beneath), it passes for me on 64 bit ubuntu but not on the 32 bit CI:
https://dev.azure.com/scipy-org/SciPy/_build/results?buildId=10968&view=logs&j=6cac4766-9adb-5b92-83ee-b21ac03a0321&t=91c05f48-75b8-58d0-e0be-5f0ceaaac72f&l=14501
can you confirm this one too @ryu577 ? 

If the values are correct it looks like we could either:
(1) leave tests running on 32 & 64 bit and drop precision to `1e-11` or `1e-12` 
or 
(b) leave `rtol` settings and mark as xfail. 

My preference would be for the former, most pvalue calculations would not require such high precision and we'd have testing for both 32 and 64. 

I suppose option (3) might be to add a branch in test to set precision based on architecture (32 vs 64 get different precision) but might be more than is necessary here.

@WarrenWeckesser + @mdhaber thoughts/opinions?

Reply:
@rlucas7 - changed the precision to 1e-11 from 1e-15. Other tests seem to pass but circleci still fails.

Reply:
For circleci, you need to merge master. (Please merge rather than rebasing, since we're in the middle of the review.) Thanks.

Reply:
> My preference would be for the former... @WarrenWeckesser + @mdhaber thoughts/opinions?

Hah, actually I'd tend toward the latter and xfail only on 32 bit (`@pytest.mark.xfail_on_32bit`). Rationale is that if on 64 bit it's still achieving 1e-15, I can't imagine changes that would make it get worse just on 32 bit. However, if we only require 1e-11 on all platforms, its possible for 64-bit to change substantially without us noticing. (but maybe this isn't credible.) I don't know. It's probably fine either way. It would be really useful if `np.testing` could offer separate 32-bit tolerances.

Reply:
> Hah, actually I'd tend toward the latter and xfail only on 32 bit (`@pytest.mark.xfail_on_32bit`). Rationale is that if on 64 bit it's still achieving 1e-15, I can't imagine changes that would make it get worse just on 32 bit.

Good point. Sorry to flip-flop here @ryu577 can you move tolerances back to `1e-15` and mark as xfail? 
 
> However, if we only require 1e-11 on all platforms, its possible for 64-bit to change substantially without us noticing (e.g. multiply all results by a factor of `1 + 1e-12` ... but maybe this isn't credible).

while this seems possible the changes seem specialized enough that it's unlikely we wouldn't notice changes that created this kind of behavior-I'm not worried about this sort of scenario.  

> I don't know. It's probably fine either way. It would be really useful if `np.testing` could offer separate 32-bit tolerances, or if there was some other easy way to do option 3.

I agree that option 3 might be best but let's get this PR merged w/the xfail markers. 
We can follow up with ideas to implement option 3 in future PR.

Reply:
Of course, just musing.

Reply:
@rlucas7  - reverted to 1e-15 and marked all tests as xfail on 32 bit.